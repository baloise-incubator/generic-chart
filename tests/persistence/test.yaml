templates:
  - deployment.yaml
#  - external-secret.yaml
#  - ingress.yaml
#  - istio-igw-withvs.yaml
  - pvc.yaml
  - service.yaml
  - serviceaccount.yaml
#  - serviceloadbalancer.yaml
#  - servicemonitor.yaml
#  - statefulset-ingress.yaml
#  - statefulset-service.yaml
#  - virtualservice.yaml
tests:
  - release:
      name: simple-app
      namespace: namespace-test
    values:
      - values.yaml
    asserts:
      - matchSnapshot: {}
