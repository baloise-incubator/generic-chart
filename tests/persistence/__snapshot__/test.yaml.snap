"":
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: generic-chart
        release: simple-app
      name: generic-chart
      namespace: namespace-test
    spec:
      replicas: 2
      revisionHistoryLimit: 1
      selector:
        matchLabels:
          app: generic-chart
          release: simple-app
      strategy:
        type: RollingUpdate
      template:
        metadata:
          labels:
            app: generic-chart
            release: simple-app
        spec:
          affinity:
            podAntiAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: app
                        operator: In
                        values:
                          - generic-chart
                      - key: release
                        operator: In
                        values:
                          - simple-app
                  topologyKey: kubernetes.io/hostname
          containers:
            - args: null
              command: null
              env: null
              envFrom: null
              image: registry.example.tld/org/repo:v1
              imagePullPolicy: IfNotPresent
              livenessProbe: null
              name: generic-chart
              ports:
                - containerPort: 8080
                  name: http
                  protocol: TCP
              readinessProbe: null
              resources: null
              securityContext: null
              startupProbe: null
              volumeMounts:
                - mountPath: /app/data
                  name: data
          imagePullSecrets: null
          initContainers: null
          nodeSelector: null
          securityContext: null
          serviceAccountName: generic-chart
          tolerations: null
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app: generic-chart
                  release: simple-app
              maxSkew: 1
              topologyKey: null
              whenUnsatisfiable: DoNotSchedule
          volumes:
            - name: data
              persistentVolumeClaim:
                claimName: data
  2: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      annotations: null
      labels:
        app: generic-chart
        release: simple-app
      name: data
      namespace: namespace-test
    spec:
      accessModes:
        - ReadWriteMany
      resources:
        requests:
          storage: 100Mi
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations: null
      labels:
        app: generic-chart
        release: simple-app
      name: generic-chart
      namespace: namespace-test
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
          targetPort: http
      selector:
        app: generic-chart
        release: simple-app
      type: ClusterIP
  4: |
    apiVersion: v1
    automountServiceAccountToken: false
    kind: ServiceAccount
    metadata:
      annotations: null
      labels:
        app: generic-chart
        release: simple-app
      name: generic-chart
      namespace: namespace-test
